File: drag.py
import time

def is_prime(n):
    # Check if n is a prime number
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

start = time.perf_counter() # start timing

primes = []
candidate = 2
while len(primes) < 100:
    if is_prime(candidate):
        primes.append(candidate)
    candidate += 1

end = time.perf_counter() # stop timing
elapsed_time = end - start

print("Elapsed time: {:.6f} seconds".format(elapsed_time))

File: drag.c
#include <stdio.h>
#include <stdint.h>
#include <time.h>

int main()
{
    uint64_t primes[100] = {0}; // an array to store the first 100 prime numbers
    uint64_t candidate = 2; // the candidate prime number
    int count = 0; // the number of prime numbers found
    clock_t start = clock(); // start timing

    while (count < 100)
    {
        // check if candidate is a prime number
        int is_prime = 1;
        for (int i = 0; i < count; i++)
        {
            if (candidate % primes[i] == 0)
            {
                is_prime = 0;
                break;
            }
        }
        if (is_prime)
        {
            // candidate is a prime number, add it to the array
            primes[count] = candidate;
            count++;
        }

        candidate++;
    }

    clock_t end = clock(); // stop timing
    double elapsed_secs = (double)(end - start) / CLOCKS_PER_SEC;

    printf("Elapsed time: %.10f seconds\n", elapsed_secs);

    return 0;
}
